### DISCUSSION TITLE
Dependent start & end dates not working

### DISCUSSION THREAD

**Discussion by kjlloydie** [filamentphp/filament#2201] [Help] [answered]
I have a form with start and end dates and I'd like to ensure the end date isn't before the start date. I've set it up roughly like this:
```php
DateTimePicker::make('start_date')
    ->withoutTime()
    ->reactive()
    ->displayFormat('d/m/Y'),
DateTimePicker::make('end_date')
    ->withoutTime()
    ->reactive()
    ->displayFormat('d/m/Y')
    ->minDate(function (Closure $get) {
        return $get('start_date');
    }),
```
which seems to work the first time the form is created, but when I change _start_date_ the calendar/selector on _end_date_ disappears, like this:
<img width="355" alt="Screenshot 2022-04-22 at 2 56 50 PM" src="https://user-images.githubusercontent.com/489813/164801162-ef1fea10-385a-41c5-8916-8245cd80cdba.png">
Am I doing something wrong? It seems like having two dates fields that are dependent on each other would be a normal use case, but I couldn't find anything in the docs about how to get it to work. Incidentally, I was also going to set `maxDate` on the _start_date_ field too, but wanted to get the end_date working first.

Alternatively, is there a date range type field/plugin (i.e. like you might get on a hotel booking site) that folks are using?

**Comment by ryangjchandler** [comment]
Feels like we're missing a `wire:ignore` somewhere..

**Comment by Z3d0X** [comment, marked as answer]
Seems to be a bug, You should create an issue for this.
@kjlloydie for now i think your best bet would be to restrict it in validation `->after('start_date')`
```php
DateTimePicker::make('start_date')
    ->withoutTime()
    ->reactive()
    ->displayFormat('d/m/Y'),
    
DateTimePicker::make('end_date')
    ->withoutTime()
    ->reactive()
    ->displayFormat('d/m/Y')
    ->after('start_date'),
```

**Reply by kjlloydie** [reply]
Thanks, I've submitted #2224

**Reply by kjlloydie** [reply]
Also, as per @danharrin in #2224, having the second date be `reactive()` is unnecessary (unless something else depends upon it):
> Fixed by #2314, you don't need the second date to be `reactive()`, only fields that are _depended upon_ should be reactive, not those that are _dependant_.
